addons/mail/static/src/components/composer/composer_tests.js

```diff
@@ -2150,6 +2151,45 @@ QUnit.test('[technical] does not crash when an attachment is removed before its
     );
 });
 
+QUnit.only('clicking on the delete attachment button multiple times should not crash', async function (assert) {
+    assert.expect(0);
+
+    const unlinkPromise = makeTestPromise();
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === "unlink" && args.model === "ir.attachment") {
+                console.log('unlink rpc mocking');
+                return unlinkPromise
+            }
+            return this._super(...arguments);
+        },
+    });
+    const composer = this.env.models['mail.composer'].create();
+    await this.createComposerComponent(composer);
+    const file = await createFile({
+        content: 'hello, world',
+        contentType: 'text/plain',
+        name: 'text.txt',
+    });
+    await afterNextRender(() =>
+        inputFiles(
+            document.querySelector('.o_FileUploader_input'),
+            [file]
+        )
+    );
+    const button = document.querySelector('.o_Attachment_asideItemUnlink');
+    button.click();
+    await unlinkPromise.resolve().then(() => {button.click();});
+    await afterNextRender(() => {
+        button.click();
+    });
+    await nextAnimationFrame();
+    button.click();
+    await nextTick();
+    button.click();
```

Comment:
document.querySelector('.o_Attachment_asideItemUnlink').click();
    await Promise.resolve();
    await Promise.resolve();
    await Promise.resolve();
    await Promise.resolve();
    await Promise.resolve();
    await Promise.resolve();
    await Promise.resolve();
    await Promise.resolve();
    await Promise.resolve();
    await Promise.resolve();
    await Promise.resolve();
    document.querySelector('.o_Attachment_asideItemUnlink').click();

Reply:
But temporal tests should ideally never use any timing function, only actual promises controlled from code (not possible here) or bus event (possible here, but probably not worth it, as we know we have to guard our handlers in general).

Reply:
We're going in a direction which would prevent having this problem in general, so it's alright not having tests because we'd be far confident it will work without needing such tests.

_(As a reminder, the solution is immediate dispatch to models, and model-level code doesn't have this sync issue)_